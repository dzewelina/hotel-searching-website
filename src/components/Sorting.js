import React from 'react';
import PropTypes from 'prop-types';

const Sorting = ({ updateSort }) => (
  <div className='form-group'>
    <label>Sort by: <b>Star Rating</b></label>
    <select className='form-control col-md-1' onChange={updateSort}>
      <option value='0'>Choose...</option>
      <option value='-1'>High to Low</option>
      <option value='1'>Low to High</option>
    </select>
  </div>
);

Sorting.propTypes = {
  updateSort: PropTypes.func.isRequired
};

export default Sorting;