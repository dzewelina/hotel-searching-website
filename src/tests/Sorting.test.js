import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import renderer from 'react-test-renderer';

import React from 'react';
import Sorting from '../components/Sorting';

describe('Sorting component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Sorting updateSort={jest.fn()}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});