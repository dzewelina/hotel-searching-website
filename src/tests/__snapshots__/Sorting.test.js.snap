// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting component matches the snapshot 1`] = `
<div
  className="form-group sorting"
>
  <label>
    Sort by: 
    <b>
      Star Rating
    </b>
  </label>
  <select
    className="form-control col-md-2"
    onChange={[Function]}
  >
    <option
      value="0"
    >
      Choose...
    </option>
    <option
      value="-1"
    >
      High to Low
    </option>
    <option
      value="1"
    >
      Low to High
    </option>
  </select>
</div>
`;
